tests/src/test/java/docs/javadsl/ConsumerSettingsTest.java

```diff
@@ -25,16 +26,17 @@ public void discoverySetup() throws Exception {
         ConfigFactory.parseString(ConsumerSettingsSpec$.MODULE$.DiscoveryConfigSection())
             .withFallback(ConfigFactory.load())
             .resolve();
-    ActorSystem system = ActorSystem.create("ConsumerSettingsTest", config);
+    ClassicActorSystemProvider system = ActorSystem.create("ConsumerSettingsTest", config);
```

Comment:
I believe we should remove the deprecation on the `DiscoverySupport` instead. Users normally can't avoid that that method is chosen with a classic actor system.

Reply:
IIUC I'm suppressing the deprecation warning instead.